src/test/java/io/anserini/integration/PretokenizedIndexEndToEndTest.java

```diff
@@ -0,0 +1,70 @@
+package io.anserini.integration;
+
+import io.anserini.collection.DocumentCollection;
+import io.anserini.collection.JsonCollection;
+import io.anserini.collection.TrecCollection;
+import io.anserini.index.IndexArgs;
+import io.anserini.index.IndexCollection;
+import io.anserini.index.generator.DefaultLuceneDocumentGenerator;
+import io.anserini.search.SearchArgs;
+
+import java.io.IOException;
+import java.util.Map;
+
+
+public class PretokenizedIndexEndToEndTest extends EndToEndTest {
+
+    @Override
+    IndexArgs getIndexArgs() {
+        IndexArgs indexArgs = createDefaultIndexArgs();
+
+        indexArgs.input = "src/test/resources/sample_docs/json/collection_tokenized";
+        indexArgs.collectionClass = JsonCollection.class.getSimpleName();
+        indexArgs.generatorClass = DefaultLuceneDocumentGenerator.class.getSimpleName();
+        indexArgs.pretokenized = true;
+        indexArgs.storeRaw = true;
+
+        return indexArgs;
+    }
+
+    @Override
+    protected void setCheckIndexGroundTruth() {
+        docCount = 2;
+        documents.put("2000000", Map.of(
+                "contents", "this was ##a simple pretokeinzed test",
+                "raw","{\n" +
+                        "  \"id\" : \"2000000\",\n" +
+                        "  \"contents\" : \"this was ##a simple pretokeinzed test\"\n" +
+                        "}"));
+        documents.put("2000001", Map.of(
+                "contents", "some time extra ##vert ##ing and some time intro ##vert ##ing",
+                "raw","{\n" +
+                        "  \"id\" : \"2000001\",\n" +
+                        "  \"contents\" : \"some time extra ##vert ##ing and some time intro ##vert ##ing\"\n" +
+                        "}"
+        ));
+
+        fieldNormStatusTotalFields = 1;
+        // whitespace analyzer keeps everything, includes docid
+        // this is ##a simple pretokenized test some time extra ##vert ##ing and intro 2000000 2000001
+        termIndexStatusTermCount = 15;
+        // this is ##a simple pretokenized test some|2 time|2 extra ##vert|2 ##ing|2 and intro 2000000 2000001  ?
+        //todo: check this field?
+        termIndexStatusTotFreq = 15;
```

Comment:
I don't see why this is 15...

Reply:
not sure either... what I generally do is make sure it's indexed correctly, and then just set the value to the number that comes out...

so at least the test case is guarding against the count changing due to a bug later on...